grails-app/domain/com/unifina/domain/security/Permission.groovy

```diff
@@ -57,6 +57,8 @@ class Permission {
 	/** When does this Permission expire? null == forever valid */
 	Date endsAt
 
+	Permission parent
```

Comment:
Should there be some sort of cascade rule? What we want is that if the `parent` gets deleted, then the children get deleted automatically (instead of failing with a foreign key constraint error). Best case is that this happens on the database level. So we should define it in the domain object, and if Grails is smart we'll se a cascade delete in the generated foreign key in the migration file. If Grails is not that smart then Grails will probably attempt to handle it. But the [definition exists](https://grails.github.io/grails2-doc/2.5.x/ref/Database%20Mapping/cascade.html) for sure.

Reply:
Oh I noticed now that child permissions are being explicitly deleted in the code when permissions are revoked. That should work, although leveraging the built-in cascade functionality may be cleaner (possibly more unstable too in case of Grails, haha).

Reply:
I didn't try the built-in Grails feature since this code works and looks simple and clean enough to me. I personally think it's better to have this simple logic in explicit viewable code instead of relying on some under the hood and potentially unstable Grails mechanism.